common/src/main/java/org/corfudb/common/metrics/micrometer/MeterRegistryProvider.java

```diff
@@ -181,7 +181,9 @@ public static void registerExternalSupplier(String identifier, Supplier<String>
     }
 
     public static void unregisterExternalSupplier(String identifier) {
-        externalMetricsSuppliers.remove(identifier);
+        synchronized (externalMetricsSuppliers) {
+            externalMetricsSuppliers.remove(identifier);
+        }
```

Comment:
Looks like we need something similar for `registerExternalSupplier` as well?

Reply:
Why it doesn't need to be added there - https://github.com/CorfuDB/CorfuDB/pull/3845#discussion_r1427478244